dartagnan/src/main/java/com/dat3m/dartagnan/expression/processing/ExprSimplifier.java

```diff
@@ -165,6 +166,24 @@ public Expression visitIntCmpExpression(IntCmpExpr cmp) {
             return expressions.makeValue(cmpResult);
         }
 
+        // ------- Operations on memory objects -------
+        if (left instanceof MemoryObject lMem && right instanceof MemoryObject rMem) {
+            final boolean sameObj = lMem.equals(rMem);
+
+            final Boolean cmpResult = switch (op) {
+                case EQ -> sameObj;
+                case NEQ -> !sameObj;
+                case LT, ULT -> sameObj ? false : null;
+                case LTE, ULTE -> sameObj ? true : null;
+                default ->
+                    throw new VerifyException(String.format("Unexpected comparison operator '%s'. Missing normalization?", op));
```

Comment:
The "greater" cases are missing. Once those are added you can get rid of the default case.